host-interaction/hardware/storage/get-disk-size.yml

```diff
@@ -0,0 +1,24 @@
+rule:
+  meta:
+    name: get disk size
+    namespace: host-interaction/hardware/storage
+    author: michael.hunhoff@fireeye.com
+    scope: function
+    att&ck:
+      - Discovery::System Information Discovery [T1082]
+    references:
+      - https://github.com/LordNoteworthy/al-khaser/blob/master/al-khaser/AntiVM/Generic.cpp#L347
+    examples:
+      - al-khaser_x86.exe_:0x4343D0
+      - al-khaser_x86.exe_:0x434010
+  features:
+    - or:
+      - api: kernel32.GetDiskFreeSpace
+      - api: kernel32.GetDiskFreeSpaceEx
+      - basic block:
+        - and:
+          - api: DeviceIoControl
+          - number: 0x7405C = IOCTL_DISK_GET_LENGTH_INFO
+      - and:
+        - string: /SELECT\s+\*\s+FROM\s+Win32_LogicalDisk/
```

Comment:
Another variant that I've seen ITW is
```
SELECT * FROM Win32_DiskDrive WHERE (SerialNumber IS NOT NULL) AND (MediaType LIKE 'Fixed hard disk%')
```
which returns something like the following
```
Partitions : 2
DeviceID   : \\.\PHYSICALDRIVE0
Model      : xxxxxxxxxxxxxxxxx
Size       : 536268025360
Caption    : xxxxxxxxxxxxxxxxx
```